src/TurbulenceClosures/turbulence_closure_implementations/TKEBasedVerticalDiffusivities/catke_vertical_diffusivity.jl

```diff
@@ -228,7 +227,7 @@ end
     Jᵇᵋ = closure.minimum_convective_buoyancy_flux
     Jᵇᵢⱼ = @inbounds Jᵇ[i, j, 1]
     Jᵇ⁺ = max(Jᵇᵋ, Jᵇᵢⱼ, Jᵇ★) # selects fastest (dominant) time-scale
-    t★ = (ℓᴰ^2 / Jᵇ⁺)^(1/3)
+    t★ = cbrt(ℓᴰ^2 / Jᵇ⁺)
```

Comment:
Can you check if this fixes the issue without the need for extra `convert`?

It will be good to avoid "over converting", because this could cause us to fail to catch spurious promotion which will hurt performance (eg removing some of the benefit of using `Float32`)

Reply:
I [checked](https://github.com/CliMA/Oceananigans.jl/issues/3870#issuecomment-2444956478) but it was not enough unfortunately.

Agree that it would be good to not over convert.

Reply:
The point is just that you may lose much of the advantage of using Float32 in the first place if you throw `convert` around